classes/order/OrderDetailPack.php

```diff
@@ -62,4 +62,171 @@ class OrderDetailPackCore extends ObjectModel
+                    WHERE pa.`id_product_attribute` = ' . $p->id_pack_product_attribute . '
+                    GROUP BY pa.`id_product_attribute`, ag.`id_attribute_group`
+                    ORDER BY pa.`id_product_attribute`';
+                $combinations = Db::readOnly()->getArray($sql);
+                foreach ($combinations as $combination) {
+                    $p->name .= ' ' . $combination['group_name'] . '-' . $combination['attribute_name'];
+                    $reference = (string) $combination['attribute_reference'];
+                    if ($reference) {
+                        $p->reference = $combination['attribute_reference'];
+                    }
+                }
+            }
+            $arrayResult[] = $p;
+        }
+        return $arrayResult;
+    }
+
+    /**
+     * Returns information about pack items.
+     *
+     * @param int $idOrder
+     * @param int $idProduct
+     * @return array
+     * @throws PrestaShopException
+     */
+    public static function getPackContent($idOrder, $idProduct)
+    {
+        $idOrder = (int) $idOrder;
+        $idProduct = (int) $idProduct;
+        if (!$idProduct || !static::isFeatureActive()) {
+            return [];
+        }
+        $cacheKey = "OrderDetailPack::getPackContent($idOrder,$idProduct)";
+        if (!Cache::isStored($cacheKey)) {
+            Cache::store($cacheKey, static::retrievePackContent($idOrder, $idProduct));
+        }
+        return Cache::retrieve($cacheKey);
+    }
+
+    /**
+     * Retrieves information about pack items from database
+     *
+     * @param int $idOrder
+     * @param int $idProduct
+     * @return array
+     * @throws PrestaShopDatabaseException
+     * @throws PrestaShopException
+     */
+    protected static function retrievePackContent($idOrder, $idProduct)
+    {
+        $idOrder = (int) $idOrder;
+        $idProduct = (int) $idProduct;
+        $idOrderDetail = static::getOrderDetailId($idOrder, $idProduct);
+        $content = [];
+        $sql = (new DbQuery())
+            ->select('id_product')
+            ->select('id_product_attribute')
+            ->select('quantity')
+            ->from('order_detail_pack')
+            ->where('id_order_detail = ' . $idOrderDetail)
+            ->orderBy('id_product, id_product_attribute');
+        $result = Db::readOnly()->getArray($sql);
+        foreach ($result as $row) {
+            $content[] = [
+                'id_product' => (int) $row['id_product'],
+                'id_product_attribute' => (int) $row['id_product_attribute'],
+                'quantity' => (int) $row['quantity']
+            ];
+        }
+        return $content;
+    }
+
+    /**
+     * Get a order detail Id
+     *
+     * @param int $idOrder
+     * @param int $idProduct
+     * @return int
+     * @throws PrestaShopDatabaseException
+     * @throws PrestaShopException
+     */
+    public static function getOrderDetailId($idOrder, $idProduct)
+    {
```

Comment:
This is probably not correct. There can be multiple records in `order_detail` for the same product ID. For example when product customization is used